pytorch_lightning/trainer/data_loading.py

```diff
@@ -55,13 +55,24 @@ def init_train_dataloader(self, model):
         self.get_train_dataloader = model.train_dataloader
 
         # determine number of training batches
-        if EXIST_ITER_DATASET and isinstance(self.get_train_dataloader().dataset, IterableDataset):
+        self.is_iterable_train_dataloader = (
+            EXIST_ITER_DATASET and hasattr(self.get_train_dataloader(), 'dataset') and
```

Comment:
some changes seem similar to #789...

Reply:
Yeah, hadn't seen that. Feel free to merge #789 through and I can rebase after :)